lib/Cake/Test/Case/Utility/ValidationTest.php

```diff
@@ -148,6 +148,10 @@ public function tearDown() {
  * @return void
  */
 	public function testNotBlank() {
+		$this->assertTrue(Validation::notBlank(0));
+		$this->assertTrue(Validation::notBlank(0.0));
+		$this->assertTrue(Validation::notBlank(-0));
+		$this->assertTrue(Validation::notBlank(-0.0));
```

Comment:
@markstory Your test is testing for positive zero not negative one.
There only two ways to get minus zero value:
1) multiply negative float value by zero (-1.23 * 0)
2) multiply 0.0 by any negative value (0.0 * -1)
Minus operator does nothing on zero values so writing -0 or -0.0 gives you positive zero instead of expected negative zero!